README.md

```diff
@@ -6,3 +6,52 @@
 [![Coverage](https://codecov.io/gh/invenia/FeatureDescriptors.jl/branch/main/graph/badge.svg)](https://codecov.io/gh/invenia/FeatureDescriptors.jl)
 [![Code Style: Blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
 [![ColPrac: Contributor's Guide on Collaborative Practices for Community Packages](https://img.shields.io/badge/ColPrac-Contributor's%20Guide-blueviolet)](https://github.com/SciML/ColPrac)
+
+FeatureDescriptors.jl is an interface package for describing features used in [models](https://invenia.github.io/Models.jl/stable/), [feature engineering](https://invenia.github.io/FeatureTransforms.jl/stable/), and other data-science workflows.
+
+## Getting Started
+
+A `Descriptor` is an abstract type that provides a way to describe the properties and relationships for a collection of features at a high-level.
+`Descriptor` supertypes define some general properties, such as the required tables or value column, that can be overloaded for specific
+subtypes as required. 
+
+For example, say that some weather data is contained in `weather.csv` file but each column describes a different feature we are interested in, such as `:temperature` or `:humidity`. 
+We can define a general `Weather` `Descriptor` with subtypes `Temperature` and `Humidity` that are loaded from the same table but using the appropriate columns:
+
+```julia
+using FeatureDescriptors
+
+abstract type Weather <: Descriptor end
+
+FeatureDescriptors.tables(::Type{<:Weather}) = ["weather.csv"]
+FeatureDescriptors.value_col(::Type{<:Weather}) = :weather  # dummy
```

Comment:
What's the purpose of this?

Reply:
It's technically needed if `Weather` is to be a valid `Descriptor`, even if in practice all you need is for all the leaf-subtypes of `Weather` to define a `value_col`. So I can see how it's confusing/misleading to include it. Do you think I should remove it?

Reply:
Hmm yeah that's confusing. I'd say remove it, except I want the interface to be followed. Is there a way to address the underlying issue, only enforcing the interface for subtypes? I guess it's not clear from the package side which descriptors will be subtypes vs. supertypes. Unless we defined an intermediate subtype.

Reply:
I've opted to remove it for now in case it's confusing. Given that the subsequent examples show you how to overload it then it should be clear. If it causes issues we can revisit.